src/scanner.cc

```diff
@@ -155,13 +157,7 @@ struct Scanner {
         }
         return false;
       } else {
-        if (lexer->lookahead == '{') {
-          advance(lexer);
-          if (lexer->lookahead == '-') {
-            advance(lexer);
-            next_token_is_comment = iswspace(lexer->lookahead);
-          }
-        }
+        next_token_is_comment = two_ahead_with_whitespace("{-");
```

Comment:
Unfortunately I don't think we can use `two_ahead_with_whitespace` here because multi-line comments in Haskell don't require a space after `{-`. For example, `{-comment-}` is a valid comment in Haskell.

Reply:
ok, will check on that